gh-58749: Removes the erroneous explanation regarding the restrictions of the global statement

bombs-kim opened the pull request:
<!-- gh-issue-number: gh-58749 -->
* Issue: gh-58749
<!-- /gh-issue-number -->
* [related mail chain](https://mail.python.org/pipermail/python-ideas/2012-April/014785.html)

As pointed out by @ncoghlan , CPython does not restrict a name in a global statement from reappearing in a [for](https://docs.python.org/dev/reference/compound_stmts.html#for) target list, [class](https://docs.python.org/dev/reference/compound_stmts.html#class) definition, function definition, or [import](https://docs.python.org/dev/reference/simple_stmts.html#import) statement

Additionally, CPython does not impose the same restrictions on [with](https://docs.python.org/dev/reference/compound_stmts.html#with) statements or [except](https://docs.python.org/dev/reference/compound_stmts.html#except) clauses, or or [variable annotations](https://docs.python.org/dev/glossary.html#term-variable-annotation).

The only real restriction is that a variable cannot be used or assigned to prior to its global declaration in the scope.

A few more notes on the changes
* I’ve even omitted the part about restrictions on formal parameters, although it's a real restriction. This is because 'a variable cannot be used or assigned to prior to its global declaration in the scope' already implies that restriction.
* I’ve revised the sentence to make it more similar to the nonlocal section.  Both explain a similar concept, and I believe the nonlocal explanation is clearer and more concise.
> The nonlocal statement applies to the entire scope of a function or
> class body. A :exc:`SyntaxError` is raised if a variable is used or
> assigned to prior to its nonlocal declaration in the scope.
* I removed the sentence, “The :keyword:global statement is a declaration which holds for the entire current code block,” as it is redundant with the following explanation: “The global statement applies to the entire scope of a function or class body.”



<!-- readthedocs-preview cpython-previews start -->
----
📚 Documentation preview 📚: https://cpython-previews--126523.org.readthedocs.build/

<!-- readthedocs-preview cpython-previews end -->

bombs-kim committed:
Removes erroneous explanation of the `global` about its restrictions; a name declared as global can appear later as a target in :keyword:with statements, :keyword:except clauses, :keyword:for target lists, :keyword:class definitions, function definitions, :keyword:import statements.

ncoghlan reviewed (changes requested):
Blocking comment from the original issue discussion: the spec wording change should be accompanied by a test suite update that ensures the revised wording remains accurate.

For closing the original issue, the `nonlocal` wording also needs to be updated, but that can be done in a separate PR after the `global` change is resolved.

There's also a process question that I need to follow up before this can be merged (the level of review needed for language spec changes is generally higher than it was back when I first filed the linked issue, even when the spec change is fixing an error in the spec)

bombs-kim committed:
Change test module description to reflect the new tests
Refer to test_scope.py in the description for additional behavioural tests
Verify that syntax errors are correctly raised for improper `global`
statements
Test various name-binding scenarios for global
variables to ensure correct behavior

bombs-kim committed:
With the change in the documentation, the previous warning no longer applies.

bombs-kim committed:
Adds name to ACKS

merwok commented on Doc/reference/simple_stmts.rst:
```diff
@@ -966,25 +966,14 @@ The :keyword:`!global` statement
-The :keyword:`global` statement is a declaration which holds for the entire
-current code block.  It means that the listed identifiers are to be interpreted
-as globals.  It would be impossible to assign to a global variable without
+The :keyword:`global` causes the listed identifiers to be interpreted
```
This reads `The global`, so is missing a noun: `The global keyword`

bombs-kim replied:
Hey @merwok , thanks for the correction. I opened #126720 that reflects this feedback and handles some more mishaps.
